pkg/topology/kademlia/kademlia.go

```diff
@@ -658,6 +670,12 @@ func (k *Kad) connect(ctx context.Context, peer swarm.Address, ma ma.Multiaddr,
 		return err
 	}
 
+	if !isFull {
+		_ = k.p2p.Disconnect(peer)
+		_ = k.p2p.Disconnect(i.Overlay)
+		return addressbook.ErrPruneEntry
```

Comment:
i am not sure about returning an error from a different package here. i think that packages in general should return errors which are internal to their logic. it might be better to move `ErrPruneEntry` to the kademlia package, since it not used by any other component